### ISSUE TITLE
Enable Knative as a Serverless Framework

### ISSUE THREAD

**Issue by stobias123** [opencv/cvat#5440]
I've done some preliminary work on this, and would really love to contribute it back to the project. Please let me know if further context or documentation is required to work on this feature and have it merged into cvat. 

### Expected Behaviour
I do not want to deploy nuclio, but would rather use knative for model serving and AI tools. 

### Current Behaviour
In the current design I must use nuclio.

### Possible Solution
I've done some preliminary work on this, and my thought is to provide an alternate [LambdaGateway](https://github.com/opencv/cvat/blob/develop/cvat/apps/lambda_manager/views.py#L44) - and enable it depending on env vars or settings. 

Basically I'll just read knative service annotations to discover models and return those in the `get` or `list` functions. Calling them is pretty straightforward, you'd just use the in-cluster service url, similar to the logic for calling [nuclio functions on localhost](https://github.com/opencv/cvat/blob/develop/cvat/apps/lambda_manager/views.py#L99)

### Context
I'm a user of [kubeflow](https://github.com/kubeflow/kubeflow). Kubeflow bundles in [knative](https://knative.dev/) as a serverless framework. I don't want to deploy nuclio because I already have a serverless framework.

**Comment by nmanovic** [comment]
@stobias123 , please just send us a PR. After that, we will review it and give you recommendations on how to integrate it.

**Comment by JaraKramar** [comment]
@stobias123 any news? I am also interested in Knative as CVAT serverless framework.

**Comment by stobias123** [comment]
been a while since I looked at this.. I moved to a different labeling platform. I'll see if I can dig up my old code.
